modules/channel-twilio/src/backend/client.ts

```diff
@@ -0,0 +1,265 @@
+        direction: 'incoming',
+        type: option.type,
+        payload: {
+          type: option.type,
+          text: option.type === 'say_something' ? option.value : option.label,
+          payload: option.value
+        },
+        threadId: to,
+        target: from
+      })
+    )
+
+    return true
+  }
+
+  async handleOutgoingEvent(event: sdk.IO.Event, next: sdk.IO.MiddlewareNextCallback) {
+    if (event.type === 'text') {
+      await this.sendText(event)
+    } else if (event.type === 'file') {
+      await this.sendImage(event)
+    } else if (event.type === 'carousel') {
+      await this.sendCarousel(event)
+    } else if (event.payload.quick_replies) {
+      await this.sendChoices(event)
+    } else if (event.payload.options) {
+      await this.sendDropdown(event)
+    }
+
+    next(undefined, false)
+  }
+
+  async sendText(event: sdk.IO.Event) {
+    await this.sendMessage(event, {
+      body: event.payload.text
+    })
+  }
+
+  async sendImage(event: sdk.IO.Event) {
+    await this.sendMessage(event, {
+      body: event.payload.title,
+      mediaUrl: [event.payload.url]
+    })
+  }
+
+  async sendCarousel(event: sdk.IO.Event) {
+    for (const { subtitle, title, picture, buttons } of event.payload.elements) {
+      let body = `${title}\n\n`
+      if (subtitle) {
+        body += `${subtitle}`
+      }
+
+      const options: MessageOption[] = []
+      for (const button of buttons) {
+        if (button.type === 'open_url') {
+          options.push({
+            label: `${button.title} : ${button.url}`,
+            value: undefined,
+            type: 'url'
+          })
+        } else if (button.type === 'postback') {
+          options.push({
+            label: button.title,
+            value: button.payload,
+            type: 'postback'
+          })
+        } else if (button.type === 'say_something') {
+          options.push({
+            label: button.title,
+            value: button.text,
+            type: 'say_something'
+          })
+        }
+      }
+
+      const args = { mediaUrl: picture ? [picture] : undefined }
+      await this.sendOptions(event, body, args, options)
+    }
+  }
+
+  async sendChoices(event: sdk.IO.Event) {
+    const options: MessageOption[] = event.payload.quick_replies.map(x => ({
+      label: x.title,
+      value: x.payload,
+      type: 'quick_reply'
+    }))
+    await this.sendOptions(event, event.payload.text, {}, options)
+  }
+
+  async sendDropdown(event: sdk.IO.Event) {
+    const options: MessageOption[] = event.payload.options.map(x => ({
+      label: x.label,
+      value: x.value,
+      type: 'quick_reply'
+    }))
+    await this.sendOptions(event, event.payload.message, {}, options)
+  }
+
+  async sendOptions(event: sdk.IO.Event, text: string, args: any, options: MessageOption[]) {
+    let body = `${text}\n`
+    for (let i = 0; i < options.length; i++) {
+      const option = options[i]
+      body += `\n${i + 1}. ${option.label}`
+    }
```

Comment:
This can be a one-liner. Please make use of ES6 features as much as possible
```js
const body =`${text}\n\n${options.map(({label}, idx) => `${idx}. ${label}`).join('\n')}` 

```